ompi/mpi/fortran/mpif-h/info_get_string_f.c

```diff
@@ -0,0 +1,143 @@
+                            (MPI_Fint *info, char *key, MPI_Fint *buflen, char *value, ompi_fortran_logical_t *flag, MPI_Fint *ierr, int key_len, int value_len),
+                            (info, key, valuelen, value, flag, ierr, key_len, value_len) )
+#else
+#define ompi_info_get_string_f pompi_info_get_string_f
+#endif
+#endif
+
+
+static const char FUNC_NAME[] = "MPI_INFO_GET_STRING";
+
+/* Note that the key_len and value_len parameters are silently added
+   by the Fortran compiler, and will be filled in with the actual
+   length of the character array from the caller.  Hence, it's the max
+   length of the string that we can use. */
+
+void ompi_info_get_string_f(MPI_Fint *info, char *key, MPI_Fint *buflen,
+                            char *value, ompi_fortran_logical_t *flag, MPI_Fint *ierr,
+                            int key_len, int value_len)
+{
+    int c_ierr, ret;
+    MPI_Info c_info;
+    char *c_key = NULL;
+    OMPI_SINGLE_NAME_DECL(buflen);
+    OMPI_LOGICAL_NAME_DECL(flag);
+    opal_cstring_t *info_str;
+
+    if (OMPI_SUCCESS != (ret = ompi_fortran_string_f2c(key, key_len, &c_key))) {
+        c_ierr = OMPI_ERRHANDLER_NOHANDLE_INVOKE(ret, FUNC_NAME);
+        if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
+        return;
+    }
+
+    c_info = PMPI_Info_f2c(*info);
+
+    if (0 == *buflen) {
+        c_ierr = ompi_info_get_valuelen(c_info, c_key,
+                                       OMPI_SINGLE_NAME_CONVERT(buflen),
+                                       OMPI_LOGICAL_SINGLE_NAME_CONVERT(flag));
+        if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
+
+        if (MPI_SUCCESS == c_ierr) {
+            OMPI_SINGLE_INT_2_FINT(buflen);
+            OMPI_SINGLE_INT_2_LOGICAL(flag);
+        }
+    } else { 
+        c_ierr = ompi_info_get(c_info, c_key, &info_str,
+                               OMPI_LOGICAL_SINGLE_NAME_CONVERT(flag));
+        if (NULL != ierr) {
+            *ierr = OMPI_INT_2_FINT(c_ierr);
+        }
+        
+
+        if (MPI_SUCCESS == c_ierr) {
+            OMPI_SINGLE_INT_2_LOGICAL(flag);
+
+            /* If we found the info key, copy the value back to the
+               Fortran string (note: all Fortran compilers have FALSE ==
+               0, so just check for any nonzero value, because not all
+               Fortran compilers have TRUE == 1).  Note: use the full
+               length of the Fortran string, which means adding one to the 3rd arg 
+               to ompi_fortran_string_c2f */
+            if (*flag) {
+                if (OMPI_SUCCESS !=
+                  (ret = ompi_fortran_string_c2f(info_str->string, value, value_len + 1))) {
+                    c_ierr = OMPI_ERRHANDLER_NOHANDLE_INVOKE(ret, FUNC_NAME);
+                    if (NULL != ierr) {
+                        *ierr = OMPI_INT_2_FINT(c_ierr);
+                    }
+                }
+                *buflen = info_str->length;
+            }
+            OBJ_RELEASE(info_str);
```

Comment:
I think this line should move into the block above

Reply:
actually I think you mean below.  but its not in the right place current.

Reply:
@devreal you're right, pushed fix.